Kickstarter-iOS/DataSources/LiveStreamDiscoveryDataSource.swift

```diff
@@ -0,0 +1,91 @@
+import Library
+import LiveStream
+import Prelude
+import UIKit
+
+internal final class LiveStreamDiscoveryDataSource: ValueCellDataSource {
+  internal func load(liveStreams: [LiveStreamEvent]) {
+    self.clearValues()
+
+    sorted(liveStreamEvents: liveStreams)
+      .groupedBy(sectionTitle(forLiveStreamEvent:))
+      .forEach { title, events in
+        guard !events.isEmpty else { return }
+        self.appendRow(value: title, cellClass: LiveStreamDiscoveryTitleCell.self, toSection: title.section)
+        events.forEach { event in
+          self.appendRow(value: event, cellClass: LiveStreamDiscoveryCell.self, toSection: title.section)
+        }
+    }
```

Comment:
^^^ this is a fun lil thing. it does the work to create titles for live/upcoming/recent with live streams grouped into them. it's heavily tested too...